docs/sorting-algorithms/heap-sort.mdx

```diff
@@ -0,0 +1,326 @@
+---
+title: Heap Sort
+slug: /sorting-algorithms/heap-sort
+---
+
+# Introduction
+
+Heap Sort is a sorting algorithm which extracts the maximum or minimum element from the array and swap it with the last element of the array for sorting in ascending or descending order N times.
+It is similar to Selection Sort, the only difference is that it uses Heap data structure for finding minimum or maximum element.
+
+# Algorithm
+
+In Heap Sort, we follow the following algorithm:
+
+* First we will convert array into MaxHeap
+
+* Swap the root of the MaxHeap with the last element of the MaxHeap 
+
+* Reduce the size of MaxHeap by 1 from the last and call Heapify on the root.
+
+* Perform step 2 and step 3 N-1 times
+
+# Explanation for Heap Sort
+
+### Example 
+
+```
+ | 4 | 1 | 2 | 7 |
+
+ Converting it to MaxHeap:
+ | 7 | 4 | 2 | 1 |
+
+ Swaping first and last element of Heap and reducing the size by 1
+ | 1 | 4 | 2 || 7 |
+
+ Calling Heapify on root
+ | 4 | 1 | 2 || 7 |
+
+ Swaping first and last element of Heap and reducing the size by 1
+ | 2 | 1 || 4 | 7 |
+
+ Calling Heapify on root
+ | 2 | 1 || 4 | 7 |
+
+ Swaping first and last element of Heap and reducing the size by 1
+ | 1 || 2 | 4 | 7 |
+```
+# Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+    { label: 'Java', value: 'java', },
+  ]
+}>
+<TabItem value="cpp">
+
+```cpp
+#include <iostream>
+using namespace std;
+
+ /* Convert the Given array of elements into Max Heap where "i"th element is parent, 
+     "2*i"th element is left child and "2*(i+1)" is right child in array representation */
+void CreateMaxHeap(int A[],int n)  
+{
+    int temp=A[n],i=n;
+    while(i>1 && temp > A[i/2])   
+    {
+        A[i]=A[i/2];
+        i=i/2;
+    }
+    A[i]=temp;
+}
+
+void DisplayHeap(int A[],int size)   // Display the Sorted elements in heap after deletion
+{
+    for(int i=1;i<=size;i++)
+    {
+        cout << A[i] << " ";
+    }
+    cout <<endl;
+    
+}
+
+void DeleteHeap(int A[],int n)   // Here we delete only root element which is Max and insert into into last element of Heap Array 
+{
+    int i,j,tmp,root;
+    i=1;
+    j=2*i;
+    root=A[1];      //Swap the root element with last element in Array
+    A[1]=A[n];
+    A[n]=root;
+    while(j<n-1)
+    {
+        if(A[j+1] > A[j])   //Inorder to maintain Max heap,Elements in array is arranged
+        {
+            j=j+1;
+        }
+        if(A[j] > A[i])
+        {
+         tmp=A[i];
+         A[i]=A[j];
+         A[j]=tmp;
+         i=j;
+         j=2*i;
+        }
+        else
+        {
+            break;
+        }
+        
+    }
+}
+
+int main()
+{
+    int A[8]={0,30,25,40,10,50,20,5};
```

Comment:
Can you have dynamic input here? I understand that the code is from NeoAlgo but we want to be consistent all across.

Reply:
I have changed the entire code of heap sort in cpp